src/hitable.h

```diff
@@ -28,4 +30,157 @@ class Hitable
 };
 
 
+/* === INSTANCING SYSTEM === */
+
+
+/**
+ * Translate Class
+ */
+class Translate : public Hitable
+{
+
+private:
+    Hitable *ptr;
+    Vec3 offset;
+
+public:
+    /**
+     * Assignment constructor to store the values of the hitable object
+     * and the transformation information.
+     *
+     * @param p Hitable object.
+     * @param displacement Transformation information (offset the initial object location)
+     */
+    Translate(Hitable *p, const Vec3 &displacement): ptr(p), offset(displacement) {}
+
+    virtual bool hit(const Ray &r, float t_min, float t_max, HitRecord &rec) const;
+    virtual bool bounding_box(float t0, float t1, AABB &box) const;
+
+};
+
+
+bool Translate::hit(const Ray &r, float t_min, float t_max, HitRecord &rec) const
+{
+    Ray moved_r(r.origin() - offset, r.direction(), r.time());
+
+    if (ptr->hit(moved_r, t_min, t_max, rec))
+    {
+        rec.p += offset;
+        return true;
+    }
+
+    return false;
+}
+
+
+bool Translate::bounding_box(float t0, float t1, AABB &box) const
+{
+    if (ptr->bounding_box(t0, t1, box))
+    {
+        box = AABB(box.min() + offset, box.max() + offset);
+        return true;
+    }
+
+    return false;
+}
+
+
+/**
+ * Rotate Y Class
+ */
+class RotateY : public Hitable
+{
+
+private:
+    Hitable *ptr;
+    float sin_theta;
+    float cos_theta;
+    bool hasbox;
+    AABB bbox;
+
+public:
+    /**
+     * Rotation Constructor.
+     * Evaluate the rotation in terms of sin and cos for the specified angle.
+     * Also evaluate the new BBOX for the rotated object.
+     *
+     * @param p
+     * @param angle
+     */
+    RotateY(Hitable *p, float angle);
+
+    virtual bool hit(const Ray &r, float t_min, float t_max, HitRecord &rec) const;
+    virtual bool bounding_box(float t0, float t1, AABB &box) const { box = bbox; return hasbox; }
+
+};
+
+
+RotateY::RotateY(Hitable *p, float angle) : ptr(p)
+{
+    auto radians = static_cast<float>((M_PI / 180.0f) * angle);
+
+    sin_theta = std::sin(radians);
+    cos_theta = std::cos(radians);
+    hasbox = ptr->bounding_box(0, 1, bbox);
+
+    auto min = Vec3(FLT_MAX, FLT_MAX, FLT_MAX);
```

Comment:
Visual Studio doesn't support the "new" `numerical_limits` evaluation procedure for compiler-specific value that a certain type allows.

So I needed to fallback to `<cfloat>` (sadly).

For more in-depth info, look at http://en.cppreference.com/w/cpp/types/climits.

Reply:
Thanks for the explanation!